src/main/java/seedu/address/ui/CommandBox.java

```diff
@@ -48,19 +50,37 @@ private void handleKeyPress(KeyEvent keyEvent) {
             // As up and down buttons will alter the position of the caret,
             // consuming it causes the caret's position to remain unchanged
             keyEvent.consume();
-
             navigateToPreviousInput();
             break;
         case DOWN:
             keyEvent.consume();
             navigateToNextInput();
             break;
+        case TAB:
+            keyEvent.consume();
+            selectClosestResultBasedOnTextFieldValue();
+            break;
         default:
             // let JavaFx handle the keypress
         }
     }
 
     /**
+     * Updates the text field with the command that is the closest to the current text field string
+     */
+    private void selectClosestResultBasedOnTextFieldValue() {
+        String currentText = commandTextField.getText().toLowerCase();
+        if (currentText.length() != 0) {
+            for (String commandWord : commandList) {
+                if (currentText.length() <= commandWord.length()
+                        && currentText.equals(commandWord.substring(0, currentText.length()))) {
```

Comment:
This what is referred to as [arrowhead code](https://blog.codinghorror.com/flattening-arrow-code/) .
You can change 
```java
if (currentText.length() != 0) {
    blah blah...
} 
```
to

```java 
if (currentText.isEmpty()) {
    return;
} 
blah blah
```
Furthermore the if statement within the for loop can be refactored into a separate method.